HackIllinois/DataSources/HIProfileDataSource.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  HIProfileDataSource.swift
+//  HackIllinois
+//
+//  Created by HackIllinois Team on 2/25/21.
+//  Copyright © 2021 HackIllinois. All rights reserved.
+//  This file is part of the Hackillinois iOS App.
+//  The Hackillinois iOS App is open source software, released under the University of
+//  Illinois/NCSA Open Source License. You should have received a copy of
+//  this license in a file with the distribution.
+//
+
+import Foundation
+import CoreData
+import HIAPI
+import os
+
+final class HIProfileDataSource {
+
+    // Serializes access to isRefreshing.
+    private static let isRefreshineQueue = DispatchQueue(label: "org.hackillinois.org.hi_profile_data_source.is_refreshing_queue", attributes: .concurrent)
```

Comment:
"isRefreshingQueue" instead of "isRefreshineQueue" (I know there was a typo in the other files too but we could fix it in this one oops)